src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -26,52 +26,70 @@ public static class ScriptJwtBearerExtensions
-                                    // TODO: DI (FACAVAL) This will be fixed once the permanent fix is in place
-                                    if (_specialized == 0 && !SystemEnvironment.Instance.IsPlaceholderModeEnabled() && Interlocked.CompareExchange(ref _specialized, 1, 0) == 0)
-                                    {
-                                        o.TokenValidationParameters = CreateTokenValidationParameters();
-                                    }
+                        // Temporary: Tactical fix to address specialization issues. This should likely be moved to a token validator
+                        // TODO: DI (FACAVAL) This will be fixed once the permanent fix is in place
+                        if (_specialized == 0 && !SystemEnvironment.Instance.IsPlaceholderModeEnabled() && Interlocked.CompareExchange(ref _specialized, 1, 0) == 0)
+                        {
+                            o.TokenValidationParameters = CreateTokenValidationParameters(GetValidAudiences());
+                        }
 
-                                    return Task.CompletedTask;
-                                },
-                                OnTokenValidated = c =>
-                                {
-                                    c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
-                                    {
-                                        new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
-                                    }));
+                        return Task.CompletedTask;
+                    },
+                    OnTokenValidated = c =>
+                    {
+                        c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
+                        {
+                            new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
+                        }));
 
-                                    c.Success();
+                        c.Success();
 
-                                    return Task.CompletedTask;
-                                }
-                            };
+                        return Task.CompletedTask;
+                    }
+                };
 
-                            o.TokenValidationParameters = CreateTokenValidationParameters();
+                o.TokenValidationParameters = CreateTokenValidationParameters(GetValidAudiences());
+
+                // TODO: DI (FACAVAL) Remove this once the work above is completed.
+                if (!SystemEnvironment.Instance.IsPlaceholderModeEnabled())
+                {
+                    // We're not in standby mode, so flag as specialized
+                    _specialized = 1;
+                }
+            });
+
+        private static string[] GetValidAudiences()
+        {
+            if (SystemEnvironment.Instance.IsPlaceholderModeEnabled() &&
```

Comment:
There's a check on line 44 that tests whether we're specialized. I think you should factor that into a private method IsSpecialized() that can be used in both places, so we're doing the check consistently.

Reply:
`Line 44` looks like some special case tactical fix. 

`_specialized == 0` --> Not specialized according to this method

`!SystemEnvironment.Instance.IsPlaceholderModeEnabled()` --> SystemEnvironment says it's specialized though.

I don't think it would make sense to mix them both in a single method :D

Reply:
Comment makes it obvious too

`// Temporary: Tactical fix to address specialization issues. This should likely be moved to a token validator`